membership-attribute-service/app/models/AccountDetails.scala

```diff
@@ -96,41 +97,40 @@ object AccountDetails {
       }
 
-      def externalisePlanName(plan: RatePlan): Option[String] = plan.product match {
-        case _: Product.Weekly => if (plan.name.contains("Six for Six")) Some("currently on '6 for 6'") else None
-        case _: Product.Paper => Some(plan.name.replace("+", " plus Digital Subscription"))
+      def externalisePlanName(plan: RatePlan): Option[String] = plan.product(catalog) match {
+        case _: Product.Weekly => if (plan.name(catalog).contains("Six for Six")) Some("currently on '6 for 6'") else None
+        case _: Product.Paper => Some(plan.name(catalog).replace("+", " plus Digital Subscription"))
         case _ => None
       }
 
+      def maybePaperDaysOfWeek(plan: RatePlan) = {
+        val dayNames = for {
+          charge <- plan.ratePlanCharges.list.toList
+            .filterNot(_.pricing.isFree) // note 'Echo Legacy' rate plan has all days of week but some are zero price, this filters those out
+          catalogZuoraPlan <- catalog.productRatePlans.get(plan.productRatePlanId)
+          dayName <- catalogZuoraPlan.productRatePlanCharges
+            .get(charge.productRatePlanChargeId)
+            .collect { case benefit: PaperDay => DayOfWeek.of(benefit.dayOfTheWeekIndex).getDisplayName(TextStyle.FULL, Locale.ENGLISH) }
+        } yield dayName
+
+        if (dayNames.nonEmpty) Json.obj("daysOfWeek" -> dayNames) else Json.obj()
+      }
+
       def jsonifyPlan(plan: RatePlan) = Json.obj(
         "name" -> externalisePlanName(plan),
         "start" -> plan.start,
         "end" -> plan.end,
         // if the customer acceptance date is future dated (e.g. 6for6) then always display, otherwise only show if starting less than 30 days from today
         "shouldBeVisible" -> (subscription.acceptanceDate.isAfter(now) || plan.start.isBefore(now.plusDays(30))),
-        "chargedThrough" -> plan.chargedThrough,
-        "price" -> plan.charges.price.prices.head.amount * 100,
-        "currency" -> plan.charges.price.prices.head.currency.glyph,
-        "currencyISO" -> plan.charges.price.prices.head.currency.iso,
-        "billingPeriod" -> plan.charges.billingPeriod.noun,
-        "features" -> plan.features.map(_.code.get).mkString(","),
-      ) ++ (plan.charges match {
-        case paperCharges: PaperCharges =>
-          Json.obj(
-            "daysOfWeek" ->
-              paperCharges.dayPrices
-                .filterNot(_._2.isFree) // note 'Echo Legacy' rate plan has all days of week but some are zero price, this filters those out
-                .keys
-                .toList
-                .map(_.dayOfTheWeekIndex)
-                .sorted
-                .map(DayOfWeek.of)
-                .map(_.getDisplayName(TextStyle.FULL, Locale.ENGLISH)),
-          )
-        case _ => Json.obj()
-      })
-
-      val sortedPlans = subscription.plans.list.sortBy(_.start.toDate)
+        "chargedThrough" -> plan.chargedThroughDate,
+        "price" -> plan.chargesPrice.prices.head.amount * 100,
```

Comment:
FIXME for some reason this has a rounding error that it didn't have in the pre-pr version
![image](https://github.com/user-attachments/assets/9ee2c298-7ec6-4e1c-8c67-13855e594dd9)

Reply:
fixed by adding .toInt in the two places that it converts to pence